### PR TITLE
Assignment4

### PR DESCRIPTION
**Pull request by reginalopezentrena**

### CODE REVIEW

File: Assignment4/reginalopezentrena-02578881/task07.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+"""**TASK 7.1: List all subclasses of "LivingThing" with RDFLib and SPARQL**"""
+
+# TO DO
+from rdflib.plugins.sparql import prepareQuery
+ns = Namespace("http://somewhere#")
+
+print("sparql query")
+q1 = prepareQuery('''
+  SELECT ?Subject WHERE {
+    ?Subject RDFS:subClassOf ns:LivingThing.
```
Comment (dgarijo) [inline Assignment4/reginalopezentrena-02578881/task07.py:33]: Does not return subclasses of subclasses of livint thing. Use `*`

File: Assignment4/reginalopezentrena-02578881/task07.py [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+  ''',
+  initNs = { "RDFS": RDFS, "ns":ns}
+)
+
+# Visualize the results
+for r in g.query(q1):
+  print(r.Subject)
+
+print("RDFLIB")
+for s, p, o in g.triples((None, RDFS.subClassOf, ns.LivingThing)):
```
Comment (dgarijo) [inline Assignment4/reginalopezentrena-02578881/task07.py:44]: Same as above. Make it a recursive function

File: Assignment4/reginalopezentrena-02578881/task07.py (hunk 1 above)
Comment (dgarijo) [inline Assignment4/reginalopezentrena-02578881/task07.py:33]: Does not return subclasses of subclasses of livint thing. Use `*`

File: Assignment4/reginalopezentrena-02578881/task07.py [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+for r in g.query(q2):
+  print(r.Individual)
+
+print("RDFLIB")
+for s, p, o in g.triples((None, RDF.type, ns.Person)):
+  print(s)
+
+for s1, p1, o1 in g.triples((None, RDFS.subClassOf, ns.Person)):
+  for s2, p2, o2 in g.triples((None, RDF.type, s1)):
+    print(s2)
```
Comment (dgarijo) [inline Assignment4/reginalopezentrena-02578881/task07.py:77]: same as above

File: Assignment4/reginalopezentrena-02578881/task07.py [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+   SELECT ?Subject (COUNT(?known) AS ?count)
+  WHERE {
+  ?Subject foaf:knows ?known.
+  }
+  GROUP BY ?Subject
+  HAVING (COUNT(?known) >= 2)
+  ''',
+  initNs = {"foaf":FOAF}
+)
+
```
Comment (dgarijo) [inline Assignment4/reginalopezentrena-02578881/task07.py:158]: missing 7.4 and 7.5 in rdflib

File: Assignment4/reginalopezentrena-02578881/task08.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+from rdflib.namespace import RDF, RDFS
+ns = Namespace("http://somewhere#")
+vcard = Namespace("http://www.w3.org/2001/vcard-rdf/3.0#")
+
+g1.add((ns.HarryPotter, vcard.EMAIL, Literal("hpotter@hogwarts.org")))
+g1.add((ns.JohnDoe, vcard.GIVE, Literal("John")))
+g1.add((ns.JohnSmith, vcard.FAMILY, Literal("Smith")))
+g1.add((ns.SaraJones, vcard.GIVEN, Literal("Sara")))
+g1.add((ns.SaraJones, vcard.FAMILY, Literal("Jones")))
+g1.add((ns.SaraJones, vcard.EMAIL, Literal("sara.jones@data.org")))
```
Comment (dgarijo) [inline Assignment4/reginalopezentrena-02578881/task08.py:38]: !!! You have to do it automatically, not by hand!

File: Assignment4/reginalopezentrena-02578881/task09.py [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+# Convierte la URI en un término URIRef
+ns1 = URIRef(uri1)
+ns2 = URIRef(uri2)
+ns3 = URIRef(uri3)
+
+ns = Namespace("http://somewhere#")
+
+g3.add((ns.JohnDoe, OWL.sameAs, ns1))
+g3.add((ns.HarryPotter, OWL.sameAs, ns2))
+g3.add((ns.SaraJones, OWL.sameAs, ns3))
```
Comment (dgarijo) [inline Assignment4/reginalopezentrena-02578881/task09.py:48]: Same as previous exercise

**Review by dgarijo — CHANGES_REQUESTED** [review]
Please address the issues raised in my review

**Comment by anaigmo** [comment]
Submitted and graded, no more PRs allowed

**Pull request merged**